packages/SwingSet/src/vats/comms/clist-inbound.js

```diff
@@ -27,6 +27,45 @@ export function makeInbound(state, stateKit) {
 
   // *-LocalForRemote: receiving an object/promise from a remote machine
 
+  function retireRemotePromiseID(remoteID, rpid) {
+    insistRemoteType('promise', rpid);
+    const remote = getRemote(state, remoteID);
+    const lpid = remote.fromRemote.get(rpid);
+    assert(lpid, X`unknown remote ${remoteID} promise ${rpid}`);
+    assert(
+      remote.toRemote.has(lpid),
+      X`unmapped remote ${remoteID} promise ${lpid}`,
+    );
+    const p = state.promiseTable.get(lpid);
+    assert(
+      !p.subscribers || p.subscribers.indexOf(remoteID) === -1,
+      X`attempt to retire remote ${remoteID} subscribed promise ${rpid}`,
+    );
+    remote.fromRemote.delete(rpid);
+    remote.toRemote.delete(lpid);
+    cdebug(`comms delete mapping r<->k ${remoteID} {rpid}<=>${lpid}`);
+  }
+
+  function enqueueRemotePromiseIDRetirement(remoteID, rpid) {
+    insistRemoteType('promise', rpid);
+    const remote = getRemote(state, remoteID);
+    remote.retirementQueue.push([remote.nextSendSeqNum, rpid]);
+  }
+
+  function retireAcknowledgedRemotePromiseIDs(remoteID, ackSeqNum) {
```

Comment:
Yeah, looks like this expects a number, and I think we're giving it a string. We should really add some typescript here.